pkg/buildscript/buildscript.go

```diff
@@ -14,8 +16,27 @@ type BuildScript struct {
 	raw *rawBuildScript
 }
 
-func New() (*BuildScript, error) {
-	return UnmarshalBuildExpression([]byte(emptyBuildExpression), nil)
+func init() {
+	// Guard against emptyBuildExpression having parsing issues
+	if !condition.BuiltViaCI() || condition.InActiveStateCI() {
+		err := New().UnmarshalBuildExpression([]byte(emptyBuildExpression))
+		if err != nil {
+			panic(err)
+		}
+	}
+}
+
+func Create() *BuildScript {
+	bs := New()
+	// We don't handle unmarshalling errors here, see the init function for that.
+	// Since the empty build expression is a constant there's really no need to error check this each time.
+	_ = bs.UnmarshalBuildExpression([]byte(emptyBuildExpression))
+	return bs
+}
+
+func New() *BuildScript {
```

Comment:
I've refactored buildscripts such that the only entry points are New, Create and Unmarshal. The build expression specific marshalling has to go through New() now, which allows us to set the AtTime on the buildscript level rather than force it into `unmarshalBuildExpression()`, which was a bit awkward.

Mainly this sets us up for retaining context when working with buildscripts, though more work is still warranted.

This came out of the original approach of using custom marshallers in our buildscript to handle ingredient functions. That approach was cancelled and revised into what you see in this PR now, but architectural improvements such as this one felt worth keeping.